firmware/Makefile

```diff
@@ -2,6 +2,9 @@ PLATFORM =
 PROJECT = 
 
 BUILD_DIR = build/$(PROJECT)/$(PLATFORM)
+CUBEMX_DIR := projects/$(PROJECT)/platforms/$(PLATFORM)/cubemx
+
+KEEP_FILES := .gitignore CMakeLists.txt README board_config.ioc generate.mk
```

Comment:
we shouldn't rely on a global setting for these files. Since the `cubemx/.gitignore` file already describes what is stored in version control, you should use its contents instead

The [`git check-ignore`](https://git-scm.com/docs/git-check-ignore) command would be excellent for this